ci-operator/step-registry/baremetalds/e2e/upgrade/test/baremetalds-e2e-upgrade-test-commands.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ baremetalds e2e upgrade test command ************"
+#export TEST_PROVIDER=packet
+
+if [[ -n "${TEST_OPTIONS}" ]]; then
+    TEST_ARGS="--options=${TEST_OPTIONS}"
+fi
+
+if [[ -n "${TEST_SKIPS}" ]]; then
+    TESTS="$(openshift-tests "${TEST_COMMAND}" --dry-run "${TEST_SUITE}")"
+    echo "${TESTS}" | grep -v "${TEST_SKIPS}" >/tmp/tests
+    echo "Skipping tests:"
+    echo "${TESTS}" | grep "${TEST_SKIPS}"
+    TEST_ARGS="${TEST_ARGS:-} --file /tmp/tests"
+fi
+
+openshift-tests "${TEST_COMMAND}" "${TEST_SUITE}" ${TEST_ARGS:-} \
+#    --provider "${TEST_PROVIDER}" \
+    -o /tmp/artifacts/e2e.log \
+    --junit-dir /tmp/artifacts/junit
```

Comment:
It seems that such logic should be merged within the existing [baremetalds-e2e-test command](https://github.com/openshift/release/blob/3b5edf83c7f3426cd2e85be4c6eae361d9f2057f/ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh#L35-L45), which already performs some work for filtering and reporting results (probably just reusing `${TEST_COMMAND}` could be enough). 
In this way, during the upgrade workflow, only the upgrade tests will be executed by the workflow test step, without the need of an additional dedicated step.

Reply:
Yeah, it definitely should be merged with the existing baremetalds-e2e-test command at some point.
As you indicated before here:
https://github.com/openshift/release/pull/11436/files#r489434008
baremetal-e2e-test command currently does not support packet ATM.

Reply:
I think in that comment https://github.com/openshift/release/pull/11436/files#r489434008 I was referring another step (`openshift-e2e-test`) defined outside the baremetalds workflow.

The `baremetads-e2e-test` instead it's the step currently being executed by the baremetalds workflow, and it's one that will require to be modified to accommodate upgrade tests execution

Reply:
Sorry. I guess I misread that.
For some reason I was thinking of `openshift-e2e-test` while you referring to `baremetalds-e2e-test`